src/app/portlets/content-types/fields/content-type-fields-drop-zone/content-type-fields-drop-zone.component.ts

```diff
@@ -41,16 +49,106 @@ export class ContentTypeFieldsDropZoneComponent {
     }
 
     /**
-     * Emit the saveField event, this method is call when a Field is dropped into the content type drop zone
+     * Emit the saveField event
+     * @param {Field} fieldToSave
+     * @memberof ContentTypeFieldsDropZoneComponent
      */
-    saveFields(): void {
-        this.saveFieldsEvent.emit(this.getFields());
+    saveFields(fieldToSave: Field): void {
+        let fields = this.getFields();
+        // Needs a better implementation
+        fields.map(field => {
+            if (!field.id && !(fieldToSave.clazz === TAB_DIVIDER.clazz || fieldToSave.clazz === LINE_DIVIDER.clazz)) {
+                field = Object.assign(field, fieldToSave);
+            } else if (field.id === this.formData.id) {
+                field = Object.assign(field, fieldToSave);
+            }
+
+            return field;
+        });
+
+        this.saveFieldsEvent.emit(fields);
+        this.cancelDialog();
+    }
+
+    /**
+     * Get the field to be edited
+     * @param {Field} fieldToEdit
+     * @memberof ContentTypeFieldsDropZoneComponent
+     */
+    editField(fieldToEdit: Field): void {
+        let fields = this.getFields();
+        // Needs a better implementation
+        fields.map((field) => {
```

Comment:
You don't need a ```map``` here, ```map``` will return a new array of whatever you return, but in this case you are only looking for a match, I think you can use ```forEach``` instead.